### PR TITLE
feat(app-shell): allow to enable StrictMode through shell

### LINKED ISSUE #3286: Add support for `StrictMode` through the `ApplicationShell`
**Is your feature request related to a problem? Please describe.**

The App Kit so far does not natively support [StrictMode](https://react.dev/reference/react/StrictMode). Applications being built with it have to "opt-into" `StrictMode` within the application's code. Consumer would achieve this by wrapping the `ApplicationShell` within a `StrictMode` element as in:

```jsx
const EntryPoint = () => (
  <StrictMode>
    <ApplicationShell
      render={() => (
        <Application />
      )}
    />
  </StrictMode>
);
```

**Describe the solution you'd like**

`StrictMode` itself can be more natively supported by the `ApplicationShell` using a `prop` which then conditionally wraps the entire tree into `<StrictMode/>`. This would allow App Kit to possibly perform other optimizations in `StrictMode` in the future.

```jsx
const EntryPoint = () => (
  <ApplicationShell
    enableStrictMode
    render={() => (
      <Application />
    )}
  />
);
```

This approach would also help to in the future enable StrictMode by default for Custom Applications by flipping the default of `enableStrictMode` to `true` while it's `false` initially. Then as another step one could remove `enableStrictMode` entirely and add a `disableStrictMode` for legacy applications.

**Describe alternatives you've considered**

`StrictMode` is designed to be applied to parts of an application. While you can not opt-out of it again if a parent opted-in. This means that applications not fully compatible with StrictMode can't use the prop but have to work their component tree upwards to support StrictMode everywhere.

**Additional context**

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

Closes #3286

### CODE REVIEW

File: packages/application-shell/src/components/application-shell/application-shell.tsx [hunk 1]
```diff
@@ -41,10 +46,24 @@ type TApplicationShellProps = {
   onRegisterErrorListeners?: (args: { dispatch: Dispatch }) => void;
   onMenuItemClick?: (event: SyntheticEvent<HTMLAnchorElement>) => void;
   disableRoutePermissionCheck?: boolean;
+  enableStrictMode?: boolean;
```
Comment (emmenko) [inline packages/application-shell/src/components/application-shell/application-shell.tsx:49]: Nit: wondering if we should be more explicit that is related to React.

```suggestion
  enableReactStrictMode?: boolean;
```

**Review by emmenko — COMMENTED** [review]
Thanks! 💯

**Comment by emmenko** [comment]
PS: we should probably add it to the `CustomViewShell` as well.

File: packages/application-shell/src/components/custom-view-shell/custom-view-shell.tsx [hunk 2]
```diff
@@ -160,19 +170,23 @@ function CustomViewShell(props: TCustomViewShellProps) {
...
   if (process.env.NODE_ENV === 'development' && !props.disableDevHost) {
     return (
-      <Suspense fallback={<ApplicationLoader />}>
-        <CustomViewDevHost applicationMessages={props.applicationMessages}>
-          <CustomViewShell applicationMessages={props.applicationMessages}>
-            {props.children}
-          </CustomViewShell>
-        </CustomViewDevHost>
-      </Suspense>
+      <StrictModeEnablement enableReactStrictMode={props.enableReactStrictMode}>
```
Comment (tdeekens) [inline packages/application-shell/src/components/custom-view-shell/custom-view-shell.tsx:173]: /cc @CarlosCortizasCT hope I could the wrapping at the right level.

**Review by emmenko — APPROVED** [review]
💯

**Review by CarlosCortizasCT — APPROVED** [review]
Thanks! 🙇

**Pull request merged**